### PR TITLE
build:created database schema for users

### PR DESCRIPTION
**Pull request by epun21403**
# Pull Request Template

## Description

Description: 
Added Prisma as a dependency and created the schema for users. 
Added "DATABASE_URL" to env file in the following format example:
postgresql://USER:PASSWORD@HOST:PORT/DATABASE?schema=SCHEMA
Here's a short explanation of each component:
-USER: The name of your database user
-PASSWORD: The password for your database user
-HOST: The name of your host name (for the local environment, it is localhost)
-PORT: The port where your database server is running (typically 5432 for PostgreSQL)
-DATABASE: The name of the [database](https://www.postgresql.org/docs/12/manage-ag-overview.html)
-SCHEMA: The name of the [schema](https://www.postgresql.org/docs/12/ddl-schemas.html) inside the database

I ran the following line in the terminal to create the tables and database:
npx prisma migrate dev --name init

## Type of Change

Please label your pull request as appropriate for the type of change it makes and remove labels that are not relevant below:

- build: affects the application build process or repository

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.md) for this project.
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Reviewers

@afrodynamic 

## Additional Notes

Please add any additional notes or context about this pull request here.

### CODE REVIEW

**Review by afrodynamic — CHANGES_REQUESTED** [review]
Make sure to either delete the `migrations` folder or add it to `.gitignore`, these files don't need to be included in the repo for our usage. The rest LGTM, good work!

```
prisma/migrations/
prisma/migrations/migration_lock.toml
prisma/migrations/20230612195154_init/
prisma/migrations/20230612195154_init/migration.sql
```

**Review by afrodynamic — APPROVED** [review]
LGTM

**Comment by epun21403** [comment]
6edd5c0
Added controllers and API routes for users table and favorites table
Added POST for users tables
Added POST, GET, and DELETE for favorites table

File: server/controllers/favoritesController.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    response.status(201).json(favorite);
+  } catch (error) {
+    console.error('Database error :', error);
+    response.status(500).send('Database error');
+  }
+};
+
+export const removePlantFromFavorite = async (request: Request, response: Response) => {
+  const { plantId } = request.params;
```
Comment (afrodynamic) [inline server/controllers/favoritesController.ts:38]: I think we should keep the request format consistent here, the other controller functions are using only using `request.body` so I'd change it here as such:
 
```suggestion
  if (!request.body || !request.body.plantId || !request.body.userId) {
    response.status(500).send("Not all information included");
    return;
  }
```
I'll add an additional suggestion where you reference plantId that'd need to be changed too

File: server/controllers/favoritesController.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  try {
+    const deletePlant = await prisma.favorites.deleteMany({
+      where: {
+        plantId: Number(plantId),
```
Comment (afrodynamic) [inline server/controllers/favoritesController.ts:43]: As noted above:

```suggestion
        plantId: Number(request.body.plantId),
        
```

File: server/routes/favoritesRoutes.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import express from 'express';
+
+import { getPlantsFromFavorite, addPlantToFavorite, removePlantFromFavorite } from '../controllers/favoritesController';
+
+const router = express.Router();
+
+router.get('/', getPlantsFromFavorite);
+router.post('/', addPlantToFavorite);
+router.delete('/:plantId', removePlantFromFavorite);
```
Comment (afrodynamic) [inline server/routes/favoritesRoutes.ts:9]: For the consistency changes you would want to remove the variable param here too:

```suggestion
router.delete('/', removePlantFromFavorite);
```

**Review by afrodynamic — CHANGES_REQUESTED** [review]
Left some suggestions to maintain consistency, the rest LGTM

**Comment by epun21403** [comment]
> Left some suggestions to maintain consistency, the rest LGTM

Implemented recommended changes to maintain consistency.

**Review by afrodynamic — APPROVED** [review]

**Review by afrodynamic — APPROVED** [review]
LGTM

**Review by afrodynamic — APPROVED** [review]

File: client/src/features/favoritePlants/FavoritePlants.tsx
Comment (jnovak79) [inline client/src/features/favoritePlants/FavoritePlants.tsx:1]: Line 12, remove console.log

Line 16, remove err from console.log

File: client/src/features/plantDetails/PlantDetails.tsx
Comment (jnovak79) [inline client/src/features/plantDetails/PlantDetails.tsx:1]: Line 26, remove console.log

line 29 remove error from console.log

**Review by jnovak79 — CHANGES_REQUESTED** [review]
Some feedback to remove console.logs. Otherwise LGTM

**Review by jnovak79 — APPROVED** [review]
LGTM

**Review by afrodynamic — APPROVED** [review]
LGTM, submitting a commit to update some of the console.logs and return statement logic

**Pull request merged**